File: engine/src/main/java/com/terralite/engine/entity/ComponentType.java
package com.terralite.engine.entity;

import java.util.Objects;

public record ComponentType<T>(String name, Class<T> type) {
    public ComponentType {
        if (Objects.requireNonNull(name, "name").isBlank()) {
            throw new IllegalArgumentException("Component type name cannot be blank");
        }
        Objects.requireNonNull(type, "type");
    }

    public static <T> ComponentType<T> of(String name, Class<T> type) {
        return new ComponentType<>(name, type);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/entity/ComponentType.kt
package com.terralite.engine.entity

@JvmRecord
data class ComponentType<T>(val name: String) {
    init {
        require(name.isNotBlank()) { "Component type name cannot be blank" }
    }

    companion object {
        @JvmStatic fun <T> of(name: String): ComponentType<T> = ComponentType(name)

        // JVM compat: callers passing Class<T> as second arg — ignored, kept for binary compat
        @JvmStatic fun <T> of(name: String, @Suppress("UNUSED_PARAMETER") type: Class<T>): ComponentType<T> = ComponentType(name)
    }
}
